pkg/pfconfig/shared/shared.go

```diff
@@ -108,10 +107,90 @@ func (c SX1301Config) MarshalJSON() ([]byte, error) {
 	return json.Marshal(m)
 }
 
-// UnmarshalJSON implements json.Unmarshaler. It currently just errors because
-// unmarshaling is not supported.
-func (c *SX1301Config) UnmarshalJSON(_ []byte) error {
-	return errUnmarshalNotImplemented
+var (
+	radioKey = regexp.MustCompile("^radio_([0|1])$")
+	txLutKey = regexp.MustCompile("^tx_lut_([0-9]{1,2})$")
+	chanKey  = regexp.MustCompile("^chan_multiSF_([0-9]{1,2})$")
+)
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (c *SX1301Config) UnmarshalJSON(msg []byte) error {
+	var root map[string]json.RawMessage
+	if err := json.Unmarshal(msg, &root); err != nil {
+		return err
+	}
+	radioMap, txLutMap, chanMap := make(map[int]RFConfig), make(map[int]TxLUTConfig), make(map[int]IFConfig)
+	for key, value := range root {
+		if key == "lorawan_public" {
+			if err := json.Unmarshal(value, &c.LoRaWANPublic); err != nil {
+				return err
+			}
+		}
+		if key == "antenna_gain" {
+			if err := json.Unmarshal(value, &c.AntennaGain); err != nil {
+				return err
+			}
+		}
+		if key == "clksrc" {
+			if err := json.Unmarshal(value, &c.ClockSource); err != nil {
+				return err
+			}
+		}
+		if key == "lbt_cfg" {
+			if err := json.Unmarshal(value, &c.LBTConfig); err != nil {
+				return err
+			}
+		}
+		if key == "chan_Lora_std" {
+			if err := json.Unmarshal(value, &c.LoRaStandardChannel); err != nil {
+				return err
+			}
+		}
+		if key == "chan_FSK" {
+			if err := json.Unmarshal(value, &c.FSKChannel); err != nil {
+				return err
+			}
+		}
+		if match := radioKey.FindStringSubmatch(key); len(match) == 2 {
```

Comment:
Nit: Using a `strings.HasPrefix` and `fmt.Scanf` is probably more efficient than matching 3 regexes in order to get to the most common one (the IF config). Could also choose to re-order.

Reply:
Damn this is super cool. I don't need those regexes. Also the only reason for using the `if` was the regex match.
Thanks!